File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""

        shortest = len(min(strs, key=len))
        i = 0

        reference = strs[0]

        while i < shortest:
            for string in strs:
                if string[i] != reference[i]:
                    return reference[:i]
            i += 1

        return reference[:shortest]

File: 0014-longest-common-prefix/0014-longest-common-prefix.cs
public class Solution
{
    public string LongestCommonPrefix(string[] strs)
    {
        if (strs.Length == 0)
        {
            return "";
        }

        if (strs.Any(str => str.Length == 0))
        {
            return "";
        }

        string reference = strs[0];

        if (!strs.All(str => str[0] == reference[0]))
        {
            return "";
        }

        int shortest = strs.Min(s => s.Length);
        int i = 1;

        while (i < shortest && i < reference.Length)
        {
            foreach (string str in strs)
            {
                if (i >= str.Length || str[i] != reference[i])
                {
                    return reference.Substring(0, i);
                }
            }
            i++;
        }

        return reference.Substring(0, shortest);
    }
}
